src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Text;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+using Windows.Storage;
+using System.Collections.Generic;
+
+namespace Windows.Storage
+{
+	public partial class FileIO
+	{
+
+		private static async Task AppendWriteTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding, bool append)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			Encoding encodingForWriter = Encoding.UTF8;
+			switch (encoding)
+			{
+				case Streams.UnicodeEncoding.Utf8:
+					encodingForWriter = Encoding.UTF8;
+					break;
+				case Streams.UnicodeEncoding.Utf16LE:
+					encodingForWriter = Encoding.Unicode;
+					break;
+				case Streams.UnicodeEncoding.Utf16BE:
+					encodingForWriter = Encoding.BigEndianUnicode;
+					break;
+			}
+
+			using (Stream fileStream = await file.OpenStreamForWriteAsync())
+			{
+				if (append)
+				{
+					fileStream.Seek(0, SeekOrigin.End);
+				}
+				else
+				{
+					// reset file content, as UWP does
+					fileStream.SetLength(0);
+				}
+
+				using (StreamWriter streamWriter = new StreamWriter(fileStream, encodingForWriter))
+				{
+					await streamWriter.WriteAsync(contents);
+				}
+			}
+		}
+
+		public static IAsyncAction AppendTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding) => AppendWriteTextAsync(file, contents, encoding, true).AsAsyncAction();
+
+		public static IAsyncAction WriteTextAsync(IStorageFile file, string contents, Streams.UnicodeEncoding encoding) => AppendWriteTextAsync(file, contents, encoding, false).AsAsyncAction();
+
+		private static string ConvertLinesToString(IEnumerable<string> lines)
+		{
+			string output = "";
+			foreach (string line in lines)
+			{
+				output = output + line + Environment.NewLine;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4827590)

Reply:
Can it be left as it is?

Reply:
No, this rule is correct, a stringbuilder is more appropriate here.

Reply:
I think you should not try to create a big string with all the content of the file in it... instead you should write each line separately to the file.

This is a technique called "streaming" which require less memory for the same operation.

Reply:
Ok, but what is preferred? Time or memory? My version takes less CPU (less switches between user space and system space), and could be also much better from disk usage aspect (one write to file and file system tables, one transaction in file system transaction log, etc.) versus many writes. Depending of disk cache usage policy, writing whole file in one step could vastly extend SSD disk life. Also one write is better on encrypted file systems.